src/Core.tsx

```diff
@@ -117,6 +118,9 @@ export class Core extends React.Component<IModalProps, IModalState> {
     this.onConfirmAuth = this.onConfirmAuth.bind(this)
     this.disconnect = this.disconnect.bind(this)
     this.connectToWallet = this.connectToWallet.bind(this)
+    if (this.props.supportedLanguages) {
+      this.availableLanguages = this.availableLanguages.filter(availableLanguage => this.props.supportedLanguages?.includes(availableLanguage.code))
+    }
```

Comment:
I would do some changes.
1. Move this to a `setupLanguages()` method to keep it more legible,
2. Also do `availableLanguages = Object.entries(...` in this method
3. Do the `filter` only once, checking `includes` in a single iteration

Reply:
Yes, this makes a lot of sense!